src/ipmi.c

```diff
@@ -200,56 +202,66 @@ static void sensor_read_handler(ipmi_sensor_t *sensor, int err,
   plugin_dispatch_values(&vl);
 } /* void sensor_read_handler */
 
-static int sensor_list_add(ipmi_sensor_t *sensor) {
-  ipmi_sensor_id_t sensor_id;
-  c_ipmi_sensor_list_t *list_item;
-  c_ipmi_sensor_list_t *list_prev;
-
+static void sensor_get_name(ipmi_sensor_t *sensor, char *buf, int buf_len) {
```

Comment:
I find having both, `buf` and `buffer` a bit confusing. I suggest to rename `buffer` to `temp` and `buf` to `buffer`.